gateway-service-knoxtoken/src/main/java/org/apache/knox/gateway/service/knoxtoken/TokenResource.java

```diff
@@ -559,6 +569,22 @@ public Response renew(String token) {
     return resp;
   }
 
+  @DELETE
+  @Path(BATCH_REVOKE_PATH)
+  @Produces({APPLICATION_JSON})
+  public Response revokeTokens(String tokenIds) {
+    final List<String> ids = GSON.fromJson(tokenIds, List.class);
```

Comment:
I wonder if the JSON input parsing would work with `@Consumes(APPLICATION_JSON)` out of the box, without manually doing it with Gson.

Reply:
I thought the same way, however, I could not make it work this way:
```
  @DELETE
  @Path(BATCH_REVOKE_PATH)
  @Produces({APPLICATION_JSON})
  @Consumes(APPLICATION_JSON)
  public Response revokeTokens(List<String> tokenIds) {
  }
```
I always received 415 (Unsupported Media Type) every time no matter what I tried (added the required headers in the `curl` command).

Reply:
Hmm, I remember there was some arguments whether DELETE requests are allowed to have request bodies or not. Maybe the frameworks think it's not allowed therefore it rejects it.

I found this:

```
There is nothing in the specifications that prohibits the use of a body for a HTTP Delete request. However for some HTTP clients this may not be implemented and some servers may ignore it, work incorrectly or refuse the request.
```

Does it work in all major web browsers?

Reply:
It does work with our embedded Jetty. If it didn't, the previous `delete` endpoint would fail too.

Reply:
Please also note I received 415 for the new `disableTokens` and `enableTokens` too, and they are `PUT` requests.